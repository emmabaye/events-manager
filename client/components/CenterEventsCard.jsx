import React, { Component } from 'react';
import { connect } from 'react-redux';
import AdminEvent from './AdminEvent.jsx';
import Pagination from './Pagination.jsx';
import { getCenterEvents } from '../actions/centerAction';

/**
 * React component to display
 * events for a given center in
 * admin panel
 */
export class CenterEventsCard extends Component {
  componentWillMount() { //eslint-disable-line
    const { centerId } = this.props;
    this.props.dispatch(getCenterEvents(centerId, "1"));
  }

  /**
   * React's method to render react component.
   * Iterates array of events to render
   *
   * @return {object}
   */
  render() {
    if (this.props.centerEvents === undefined) {
      return null;
    }
    if (this.props.centerEvents.rows === undefined) {
      return null;
    }
    return (
      <div className="card">
        <h6 className="card-header"><b>EVENTS</b></h6>
        <div className="row">
          { this.props.centerEvents.rows !== undefined &&
            (this.props.centerEvents.rows.map( //eslint-disable-line
              (event) => <AdminEvent key={event.id} event={event} dispatch={this.props.dispatch} />))
          }
        </div>
        { this.props.centerEvents.rows.length > 0 &&
          <Pagination
            firstPage={this.props.centerEvents.page.firstPage}
            currentPage={this.props.centerEvents.page.currentPage}
            previousPage={this.props.centerEvents.page.previousPage}
            nextPage={this.props.centerEvents.page.nextPage}
            lastPage={this.props.centerEvents.page.lastPage}
            dispatch={this.props.dispatch}
            getItems={getCenterEvents}
          />
        }
      </div>
    );
  }
}

/**
 * Makes redux dispatch method available in this
 * components props
 *
 * @param  {object} dispatch dispatch method
 * @return {object} props object
 */
const mapDispatchToProps = (dispatch) => ({
  dispatch: (actionObject) => dispatch(actionObject)
});

/**
 * Makes redux state available in this
 * component's props
 *
 * @param  {object} state global state
 * @return {object} props object
 */
const mapStateToProps = (state) => ({
  centerEvents: state.centerReducer.centerEvents
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CenterEventsCard);
