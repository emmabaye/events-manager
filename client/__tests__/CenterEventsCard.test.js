import CenterEventsCard from '../components/CenterEventsCard.jsx';

describe('CenterEventsCard Component', () => {
  const initialState = {
    centerReducer: {
      allCenters: {
        status: "",
        message: "",
        data: []
      },
      status: "",
      message: "",
      center: {},
      centerEvents: {
        centerEvents: {
          rows: [{
            id: '1',
            title: 'Seminar',
            description: "",
            location: "",
            time: "5:00",
            date: '2020-01-01',
            image: "",
            Center: []
          }]
        }
      }
    }
  };

  const store = mockStore(initialState);

  const props = {

  };

  global.localStorage = {
    getItem: () => {}
  };

  describe('CenterEventsCardu  should render self', () => {
    it('it should render for smart component', () => {
      const wrapper = mount(<Provider store={store} ><CenterEventsCard {...props} /></Provider>);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should return null for undefined events rows', () => {
      const wrapper = mount(<Provider store={store} ><CenterEventsCard /></Provider>);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.length).toEqual(1);
    });

    it('it should return center events rows', () => {
      const wrapper = mount(<Provider store={store} ><CenterEventsCard /></Provider>);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.length).toEqual(1);
    });
  });
});
