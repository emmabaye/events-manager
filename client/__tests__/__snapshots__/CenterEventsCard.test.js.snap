// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterEventsCard Component CenterEventsCardu  should render self it should render for smart component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CenterEventsCard)>
    <CenterEventsCard
      centerEvents={
        Object {
          "centerEvents": Object {
            "rows": Array [
              Object {
                "Center": Array [],
                "date": "2020-01-01",
                "description": "",
                "id": "1",
                "image": "",
                "location": "",
                "time": "5:00",
                "title": "Seminar",
              },
            ],
          },
        }
      }
      dispatch={[Function]}
    />
  </Connect(CenterEventsCard)>
</Provider>
`;

exports[`CenterEventsCard Component CenterEventsCardu  should render self it should return center events rows 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CenterEventsCard)>
    <CenterEventsCard
      centerEvents={
        Object {
          "centerEvents": Object {
            "rows": Array [
              Object {
                "Center": Array [],
                "date": "2020-01-01",
                "description": "",
                "id": "1",
                "image": "",
                "location": "",
                "time": "5:00",
                "title": "Seminar",
              },
            ],
          },
        }
      }
      dispatch={[Function]}
    />
  </Connect(CenterEventsCard)>
</Provider>
`;

exports[`CenterEventsCard Component CenterEventsCardu  should render self it should return null for undefined events rows 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CenterEventsCard)>
    <CenterEventsCard
      centerEvents={
        Object {
          "centerEvents": Object {
            "rows": Array [
              Object {
                "Center": Array [],
                "date": "2020-01-01",
                "description": "",
                "id": "1",
                "image": "",
                "location": "",
                "time": "5:00",
                "title": "Seminar",
              },
            ],
          },
        }
      }
      dispatch={[Function]}
    />
  </Connect(CenterEventsCard)>
</Provider>
`;
